### PR TITLE
Fix dropdown styling

### PR DESCRIPTION
**Pull request by mmso**
Changes:

* `DropdownActions` now only supports a list of buttons, since the primary button can only be a button
* Adds support for animation
* `DropdownMenu` now only supports children, instead of receiving a list as array.
* Removes `DropdownItem` because I want to avoid generic components if they end up taking a lot of props. Replaced by `DropdownButton`, `DropdownLink`, etc., or whatever you want.

How to use:

```jsx
            <Dropdown
                align="[right|left]"
                caret
                className={'pm-button'}
                title={'Open dropdown'}
                content={'Press me'}
            >
                <DropdownMenu>
                    <DropdownButton className="alignleft">
                       First button
                    </DropdownButton>
                    <DropdownButton className="alignleft" onClick={handleClick}>
                       Second button
                    </DropdownButton>
                    <DropdownLink className="alignleft" href="http://www.foo.bar">
                       A link
                    </DropdownLink>
                </DropdownMenu>
                <PrimaryButton>Another button below the menu</PrimaryButton>
            </Dropdown>
```

Closes: #59

### CODE REVIEW

File: containers/domains/DomainAddresses.js [hunk 1]
```diff
@@ -7,7 +7,7 @@ import { Dropdown, DropdownMenu, Icon } from 'react-components';
 const DomainAddresses = ({ domain }) => {
     const addresses = domain.addresses || [];
     const title = addresses.map(({ Email }) => Email).join(', ');
-    const list = addresses.map(({ Email: text }) => ({ text }));
+    const actions = addresses.map(({ Email: text }) => text);
```
Comment (dhoko) [inline containers/domains/DomainAddresses.js]: useless alias ;)

File: containers/keys/KeysActions.js [hunk 2]
```diff
@@ -46,11 +46,13 @@ const ACTIONS_TO_TEXT = {
...
-        text: ACTIONS_TO_TEXT[action]().text,
-        type: 'button',
-        onClick: cb
-    }));
+    const list = actions
+        .map(({ action, cb }) => ({
+            text: ACTIONS_TO_TEXT[action]().text,
+            onClick: cb
+        }))
+        .filter(Boolean);
```
Comment (dhoko) [inline containers/keys/KeysActions.js]: useless filter the array is always full

File: containers/members/MemberAddresses.js [hunk 3]
```diff
@@ -7,8 +7,8 @@ import { Dropdown, DropdownMenu, Icon } from 'react-components';
 const MemberAddresses = ({ member }) => {
     const addresses = member.addresses || [];
     const title = addresses.map(({ Email }) => Email).join(', ');
-    const list = addresses.map(({ Email: text }) => ({ text }));
-    const n = list.length;
+    const actions = addresses.map(({ Email: text }) => text);
```
Comment (dhoko) [inline containers/members/MemberAddresses.js]: idem here for the alias ;)

File: components/dropdown/Dropdown.js [hunk 4]
```diff
@@ -44,16 +59,27 @@ const Dropdown = ({ isOpen, children, className, content, autoClose, autoCloseOu
...
-            <Button className={className} onClick={handleClick} aria-expanded={open}>
-                {content}
-            </Button>
-            {open ? (
-                <div className="dropDown-content" onClick={handleClickContent}>
-                    {children}
-                </div>
-            ) : null}
+        <div className={`${dropdownClassName} ${className}`} ref={wrapperRef}>
+            <button
```
Comment (EpokK) [inline components/dropdown/Dropdown.js]: I suppose you have a reason to not use the `Button` component here. Then did you check how make the dropdown looks primary if needed?

File: components/dropdown/Dropdown.js [hunk 5]
```diff
@@ -44,16 +59,27 @@ const Dropdown = ({ isOpen, children, className, content, autoClose, autoCloseOu
...
+                className="increase-surface-click"
+                aria-expanded={open}
+                onClick={handleClick}
+                type="button"
+            >
+                <span className="mauto">
+                    {content} {caretContent}
+                </span>
+            </button>
+            <div className={contentClassName} onClick={handleClickContent} hidden={!open}>
```
Comment (EpokK) [inline components/dropdown/Dropdown.js]: Why keep the content of the `Dropdown` in the page? I suppose for animation?

Comment (mmso) [reply components/dropdown/Dropdown.js]: Yes, and we can attach an event handler for `onAnimationEnd` and remove it if we see that it becomes a performance issue to always render all of them. But I think it should be ok so I didn't add it for now

**Pull request merged**